packages/kbn-eslint-plugin-eslint/rules/no_unsafe_hash.js

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+const allowedAlgorithms = ['sha256', 'sha3-256', 'sha512'];
+
+module.exports = {
+  allowedAlgorithms,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Allow usage of createHash only with allowed algorithms.',
+      category: 'FIPS',
+      recommended: false,
+    },
+    messages: {
+      noDisallowedHash:
+        'Usage of createHash with "{{algorithm}}" is not allowed. Only the following algorithms are allowed: [{{allowedAlgorithms}}]. If you need to use a different algorithm, please contact the Kibana security team.',
+    },
+    schema: [],
+  },
+  create(context) {
+    let isCreateHashImported = false;
+    let createHashName = 'createHash';
+    const sourceCode = context.getSourceCode();
+
+    const disallowedAlgorithmNodes = new Set();
+
+    function isAllowedAlgorithm(algorithm) {
+      return allowedAlgorithms.includes(algorithm);
+    }
+
+    function getIdentifierValue(node) {
+      const scope = sourceCode.getScope(node);
+      if (scope) {
+        const variable = scope.variables.find((variable) => variable.name === node.name);
+        if (variable && variable.defs.length > 0) {
+          const def = variable.defs[0];
+          if (
+            def.node.init &&
+            def.node.init.type === 'Literal' &&
+            !isAllowedAlgorithm(def.node.init.value)
+          ) {
+            disallowedAlgorithmNodes.add(node.name);
+            return def.node.init.value;
+          }
+        }
+      }
+      return undefined;
+    }
+
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value === 'crypto') {
+          node.specifiers.forEach((specifier) => {
+            if (specifier.type === 'ImportSpecifier' && specifier.imported.name === 'createHash') {
```

Comment:
__question:__ any reason we don't need\want to cover `createSign`, `createHmac`, and upcoming [`hash`](https://nodejs.org/api/crypto.html#cryptohashalgorithm-data-outputencoding)?